portal/index.php

```diff
@@ -384,10 +384,10 @@ function validate_new_pass() {
                 <div class="row">
                     <div class="col-12">
                         <?php if ($GLOBALS['portal_onsite_two_register']) { ?>
-                            <button class="btn btn-secondary float-left" onclick="location.replace('./account/register.php')"><?php echo xlt('Register'); ?></button>
+                            <button class="btn btn-secondary float-left" onclick="location.replace('./account/register.php?site=<?php echo urlencode($_SESSION['site_id']); ?>')"><?php echo xlt('Register'); ?></button>
                         <?php } ?>
                         <?php if ($GLOBALS['portal_two_pass_reset'] && isset($_GET['w']) && (isset($_GET['u']) || isset($_GET['p']))) { ?>
-                            <button class="btn btn-danger ml-2" onclick="location.replace('./index.php?requestNew=1')"><?php echo xlt('Reset Credentials'); ?></button>
+                            <button class="btn btn-danger ml-2" onclick="location.replace('./index.php?requestNew=1&site=<?php echo urlencode($_SESSION['site_id']); ?>')"><?php echo xlt('Reset Credentials'); ?></button>
```

Comment:
for the above 4 urlencode() calls, would instead use the attr_url() function

Reply:
nothing wrong with using urlencode!

Reply:
the attr_url is running urlencode (for the url encoding), but then also runs attr (for the html escaping).

as an aside, from a escaping geek perspective, this is a pretty neat use case where could actually do triple escaping (html/js/url) since this falls within a /html/js/url context. I think it may be overkill in this situation, but it would look something like:
```
<button class="btn btn-danger ml-2" onclick="location.replace('./index.php?requestNew=1&site=' + <?php echo attr_js(urlencode($_SESSION['site_id'])); ?>)"><?php echo xlt('Reset Credentials'); ?></button>
```